tests/experimental/datasources/test_pandas_s3_datasource.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import List, cast
+
+import pandas as pd
+import pytest
+from moto import mock_s3
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.util import S3Url
+from great_expectations.experimental.datasources import PandasS3Datasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    S3DataConnector,
+)
+from great_expectations.experimental.datasources.dynamic_pandas import PANDAS_VERSION
+from great_expectations.experimental.datasources.file_path_data_asset import (
+    _FilePathDataAsset,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.pandas_file_path_datasource import (
+    CSVAsset,
+)
+
+logger = logging.getLogger(__file__)
+
+
+try:
+    import boto3
+    import botocore
+    from botocore.client import BaseClient
+except ImportError:
+    logger.debug(
+        "Unable to load boto3 or botocore; install optional boto3 and botocore dependencies for support."
+    )
+    boto3 = None
+    botocore = None
+    BaseClient = None
+
+
+# apply markers to entire test module
+pytestmark = [
+    pytest.mark.skipif(
+        PANDAS_VERSION < 1.2, reason=f"ZEP pandas not supported on {PANDAS_VERSION}"
+    )
+]
+
+
+def _get_region_name() -> str:
+    return "us-east-1"
+
+
+def _get_boto3_client() -> BaseClient:
+    client = boto3.client("s3", region_name=_get_region_name())
+    return client
+
+
+def _build_pandas_s3_datasource() -> PandasS3Datasource:
```

Comment:
I could not use fixtures here, because no matter in which order I tried them, the `@fixture` decorator and the `@mock_s3` decorator would not co-exist.  Hence, I converted fixtures into the private "_build" type utility methods.

Reply:
You should be able to use `mock_s3` as a context manager.
https://docs.getmoto.org/en/latest/docs/getting_started.html#context-manager

If it were me I would probably use the context manager to define a `mock_s3` fixture which could then be combined with other fixtures (or tests).

Reply:
@Kilo59 This sounds like a good idea -- I will include the improvement you are recommending as part of the next pull request.  Thank you.